js/app/components/AddressAutosuggest/google.js

```diff
@@ -50,35 +55,95 @@ const autocompleteOptions = {
 }
 
 export const onSuggestionsFetchRequested = function({ value }) {
-
-  // https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service
-  autocompleteService.getPlacePredictions({
-    ...autocompleteOptions,
-    // https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service?hl=en#AutocompletionRequest.locationRestriction
-    locationRestriction: latLngBounds,
-    input: value,
-  }, (predictions, status) => {
-
-    if (status === window.google.maps.places.PlacesServiceStatus.OK && Array.isArray(predictions)) {
-
-      const predictionsAsSuggestions = predictions.map((result, idx) => ({
-        type: 'prediction',
-        value: result.description,
-        id: result.place_id,
-        description: result.description,
-        index: idx,
-        google: result,
-        // *WARNING*
-        // At this step, we DON'T have the lat/lng
-        // It will be obtained when selecting the suggestion
-      }))
-
+  
+  if (isNewPlacesApi) {
+    if (!sessionToken) {
+      sessionToken = uuidv4()
+    }
+    
+    axios.post(
+      'https://places.googleapis.com/v1/places:autocomplete',
+      {
+        input: value,
+        sessionToken: sessionToken,
+        locationRestriction: { "rectangle": {
+          "low" : {
+            "latitude": latLngBounds.getSouthWest().lat(),
+            "longitude": latLngBounds.getSouthWest().lng()
+          },
+          "high" : {
+            "latitude": latLngBounds.getNorthEast().lat(),
+            "longitude": latLngBounds.getNorthEast().lng()
+          },
+        }},
+        includedPrimaryTypes: ['street_address'],
+        languageCode: localeDetector()
+      },
+      {headers: {"X-Goog-Api-Key": googleApiKey}}
+    ).then(async resp => {
+      const { suggestions } = resp.data
+      let predictionsAsSuggestions
+
+      // if no suggestions are found, the API returns nothing
+      if (!suggestions) {
+        predictionsAsSuggestions = []
+      } else {
+        predictionsAsSuggestions = suggestions.map((suggestion, idx) => {
+          let prediction = suggestion.placePrediction
+          return {
+            type: 'prediction',
+            value: prediction.text.text,
+            id: prediction.placeId,
+            description: prediction.text.text,
+            index: idx,
+            google: prediction,
+            // *WARNING*
+            // At this step, we DON'T have the lat/lng
+            // It will be obtained when selecting the suggestion
+          }
+        })
+      }
       this._autocompleteCallback(predictionsAsSuggestions, value)
 
-    }
+    }).catch(error => {
+      // our API key is not valid for the places API (new), fallback to legacy API
+      if (error.response.status === 403) {
+        console.error("[Google adapter] Using legacy Places API")
```

Comment:
Maybe log to datadog instead so we can see errors and warn cooperatives?

See https://github.com/coopcycle/coopcycle-web/blob/master/js/app/hooks/useDatadog.js or `window.DatadogLogger` can be accessed without a hook if needed